src/main/java/com/adaptris/csv/aggregator/CsvAggregating.java

```diff
@@ -0,0 +1,103 @@
+package com.adaptris.csv.aggregator;
+
+import com.adaptris.annotation.AdvancedConfig;
+import com.adaptris.annotation.InputFieldDefault;
+import com.adaptris.annotation.InputFieldHint;
+import com.adaptris.core.AdaptrisMessage;
+import com.adaptris.core.CoreException;
+import com.adaptris.core.services.aggregator.MessageAggregatorImpl;
+import com.adaptris.core.util.ExceptionHelper;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.List;
+
+public class CsvAggregating extends MessageAggregatorImpl
```

Comment:
For anyone who actually reads this far, this class could do everything and it's left to the user to choose whether they care about mismatched columns.